src/newgrf_engine.cpp

```diff
@@ -704,6 +710,12 @@ static uint32 VehicleGetVariable(Vehicle *v, const VehicleScopeResolver *object,
 			return ret;
 		}
 
+		case 0x63: { // Poweredness relative to arbitrary railtype
+			if (v->type != VEH_TRAIN) return 0;
+			RailType rt = GetTileRailType(v->tile);
+			return HasPowerOnRail(object->ro.grffile->railtype_map[parameter], rt);
```

Comment:
Use ? : to define an explicit value, instead of using implicit bool->int conversion.